code/lib/types/src/modules/composedStory.ts

```diff
@@ -76,7 +76,7 @@ export interface ComposeStoryFn<TRenderer extends Renderer = Renderer, TArgs ext
   (
     storyAnnotations: AnnotatedStoryFn<TRenderer, TArgs> | StoryAnnotations<TRenderer, TArgs>,
     componentAnnotations: ComponentAnnotations<TRenderer, TArgs>,
-    projectAnnotations: ProjectAnnotations<TRenderer>,
+    projectAnnotations: FlexibleProjectAnnotations<TRenderer>,
```

Comment:
I propose you do not "leak" the `NamedOrDefault` part down so far.

Let's convert `NamedOrDefaultProjectAnnotations` to `ProjectAnnotations` as soon as we can, and not let the underlaying code worry about this.